pkg/cmd/apply.go

```diff
@@ -315,9 +339,20 @@ func runApplyUpgradeIfNeeded(s *state.State, opts *applyOpts) error {
 
 	operations := []string{}
 
-	tasksToRun := tasks.WithResources(nil)
+	var tasksToRun tasks.Tasks
```

Comment:
But what tasks to run when `upgradeNeeded || opts.ForceUpgrade` is not true? We need to run `tasks.WithResources` in that case, but it will not happen this way.

Reply:
@kron4eg nice catch, updated :+1: